packages/function/src/function.ts

```diff
@@ -1,9 +1,9 @@
 export type Guard<A, B extends A> = (value: A) => value is B
 
-export type Predicate<A> = (value: A) => boolean
+export type Predicate<A> = (value: A, index?: number) => boolean
```

Comment:
Let's not add indices here. Just use a non-aliased types in `array`

Reply:
@katis You are right, fixed